### PR TITLE
You now see the vibebot do its thing

### LINKED ISSUE #72847: vibebot is invisible
Round ID: [198594](https://scrubby.melonmesa.com/round/198594)
they glows but they spritesn't

image

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request

Fixes #72847

You now see the vibes and not just feel them

## Changelog
:cl:
fix: vibebot being invisible
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 1]
```diff
@@ -942,7 +942,12 @@ Pass a positive integer as an argument to override a bot's default speed.
 				ejectpai(usr)
 
 /mob/living/simple_animal/bot/update_icon_state()
-	icon_state = "[initial(icon_state)][get_bot_flag(bot_mode_flags, BOT_MODE_ON)]"
+	var/base_state
+	if(!isnull(base_icon_state))
+		base_state = initial(base_icon_state)
+	else
+		base_state = initial(icon_state)
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/simple_animal/bot/bot.dm:949]: When will this ever be null?

Comment (SyncIt21) [reply code/modules/mob/living/simple_animal/bot/bot.dm:949]: Just an assumption so I don't break existing behavior. For example floor bot does not use  base_icon_state and I'm not sure about other bots

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 2]
```diff
@@ -942,7 +942,12 @@ Pass a positive integer as an argument to override a bot's default speed.
 				ejectpai(usr)
 
 /mob/living/simple_animal/bot/update_icon_state()
-	icon_state = "[initial(icon_state)][get_bot_flag(bot_mode_flags, BOT_MODE_ON)]"
+	var/base_state
+	if(!isnull(base_icon_state))
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/bot/bot.dm:946]: Invert this and also probably use a ternary, shouldn't `if (!x) else` for short bodies

File: code/modules/mob/living/simple_animal/bot/vibebot.dm [hunk 3]
```diff
@@ -24,9 +25,9 @@
 
 /mob/living/simple_animal/bot/vibebot/Initialize(mapload)
 	. = ..()
-	update_appearance()
 	vibe_ability = new(src)
 	vibe_ability.Grant(src)
+	update_appearance(UPDATE_ICON_STATE)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/bot/vibebot.dm:30]: Don't do this, it breaks emissives.

```suggestion
	update_appearance(UPDATE_ICON)
```

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 4]
```diff
@@ -942,7 +942,7 @@ Pass a positive integer as an argument to override a bot's default speed.
 				ejectpai(usr)
 
 /mob/living/simple_animal/bot/update_icon_state()
-	icon_state = "[initial(icon_state)][get_bot_flag(bot_mode_flags, BOT_MODE_ON)]"
+	icon_state = "[isnull(base_icon_state)? initial(icon_state) : initial(base_icon_state)][get_bot_flag(bot_mode_flags, BOT_MODE_ON)]"
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/bot/bot.dm:945]:
```suggestion
	icon_state = "[isnull(base_icon_state )? initial(icon_state) : base_icon_state][get_bot_flag(bot_mode_flags, BOT_MODE_ON)]"
```
the point of base icon state is that its the default icon we build off. why are you using initial with it?

Comment (SyncIt21) [reply code/modules/mob/living/simple_animal/bot/bot.dm:945]: You right

**Review by jlsnow301 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 5]
```diff
@@ -942,7 +942,7 @@ Pass a positive integer as an argument to override a bot's default speed.
 				ejectpai(usr)
 
 /mob/living/simple_animal/bot/update_icon_state()
-	icon_state = "[initial(icon_state)][get_bot_flag(bot_mode_flags, BOT_MODE_ON)]"
+	icon_state = "[isnull(base_icon_state )? initial(icon_state) : base_icon_state][get_bot_flag(bot_mode_flags, BOT_MODE_ON)]"
```
[the remark this answers is not in the document]
Comment (SyncIt21) [reply code/modules/mob/living/simple_animal/bot/bot.dm:945]: > and also probably use a ternary
Ummm?

[the remark this answers is not in the document]
Comment (SyncIt21) [reply code/modules/mob/living/simple_animal/bot/bot.dm:945]: @jlsnow301??

[the remark this answers is not in the document]
Comment (jlsnow301) [reply code/modules/mob/living/simple_animal/bot/bot.dm:945]: I'm high apparently

[the remark this answers is not in the document]
Comment (jlsnow301) [reply code/modules/mob/living/simple_animal/bot/bot.dm:945]: just move the space over, it's fine, `isnull(base_icon_state) ?`

[the remark this answers is not in the document]
Comment (SyncIt21) [reply code/modules/mob/living/simple_animal/bot/bot.dm]: > I'm high apparently

Nah your probably just tired after a long day 😂  Get some rest my friend

File: code/modules/mob/living/simple_animal/bot/bot.dm (hunk 4 above)
Comment (Mothblocks) [reply code/modules/mob/living/simple_animal/bot/bot.dm]: did you place the space in the wrong place

**Pull request merged**